src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -178,7 +178,7 @@ public CSharpType GetNonNullable()
             if (!IsNullable)
                 return this;
 
-            return new CSharpType(Implementation, Namespace, Name, IsValueType, IsEnum, false, Arguments);
+            return IsFrameworkType ? new CSharpType(_type!, false, Arguments) : new CSharpType(Implementation, Namespace, Name, IsValueType, IsEnum, false, Arguments);
```

Comment:
this might be better? 
```suggestion
            return IsFrameworkType ? new CSharpType(FrameworkType, false, Arguments) : new CSharpType(Implementation, Namespace, Name, IsValueType, IsEnum, false, Arguments);
```